packages/material-ui/src/Select/Select.js

```diff
@@ -101,7 +101,7 @@ const Select = React.forwardRef(function Select(inProps, ref) {
       ...(input ? input.props.inputProps : {}),
     },
     ...(multiple && native && variant === 'outlined' ? { notched: true } : {}),
-    ref,
+    ref: input ? undefined : ref,
```

Comment:
What if developers are doing the opposite to https://codesandbox.io/s/material-demo-forked-olbo2?file=/demo.tsx? Shouldn't we support

```jsx
        <Select
          labelId="demo-customized-select-label"
          id="demo-customized-select"
          value={age}
          onChange={handleChange}
          input={
            <BootstrapInput
              startAdornment="click me"
              ref={inputRef}
              onClick={() => console.log(inputRef.current)}
            />
          }
        >
```

and 

```jsx
        <Select
          labelId="demo-customized-select-label"
          id="demo-customized-select"
          value={age}
          onChange={handleChange}
          ref={inputRef}
          input={
            <BootstrapInput
              startAdornment="click me"
              onClick={() => console.log(inputRef.current)}
            />
          }
        >
```

Reply:
This is the expected behavior and this passing of the ref causes the error to show undefined.

I think that when this component was created the intention was to be used this way, I don't see any problem with this approach.

But I don't like to receive an undefined if I put the ref in the input, but I understand the functionality of component Select to use prop ref in Select, and inputRef in the input.

I tried to use this way `ref: input?.ref !== null ? undefined : ref`, that accept the two approaches.
But this causes an error `AssertionError: expected null to be an instance of HTMLDivElement` in the describeConformance in the tests.

Reply:
bf91be014579852050f6ae7b4e1d9cf35e0a6ab7 should resolve this. `ref` is now allowed in `Select` and the `input` element.